gammapy/data/data_store.py

```diff
@@ -584,9 +585,73 @@ def make_exposure_image(self, fov, energy_range):
         """
         raise NotImplementedError
 
+    def make_psf(self, source_position, energy=None, theta=None):
+        """Make energy-dependent PSF for a given source position.
+
+        Parameters
+        ----------
+        source_position : `~astropy.coordinates.SkyCoord`
+            Coordinates of the source position for which we want to calculate the psf
+        energy : `~astropy.units.Quantity`
+            1D Energy Array on which you want to define the new `~gammapy.irf.EnergyDependentTablePSF`
+        theta : `~astropy.coordinates.Angle`
+            1D Angle Array on which you want to define the new `~gammapy.irf.EnergyDependentTablePSF` (offset in the FOV)
+
+        Returns
+        -------
+        energy_dependant_psftab : `~gammapy.irf.EnergyDependentTablePSF`
+            Energy dependent psf table
+        """
+
+        offset = source_position.separation(self.pointing_radec)
+        if not energy:
+            energy = self.psf.to_table_psf(theta=offset).energy
+        if not theta:
+            theta = self.psf.to_table_psf(theta=offset).offset
+        psf_value = self.psf.to_table_psf(theta=offset).evaluate(energy)
+        arf = self.aeff.evaluate(offset=offset, energy=energy).to("cm2")
+        exposure = arf * self.observation_live_time_duration
+        psf = EnergyDependentTablePSF(energy=energy, offset=theta, exposure=exposure,
+                                                          psf_value=psf_value)
+        return psf
+
 
 class ObservationList(list):
     """List of `~gammapy.data.DataStoreObservation`
 
     Could be extended to hold a more generic class of observations
     """
+
+    def make_psf(self, source_position, energy=None, theta=None):
+        """Make energy-dependent PSF for a given source position and a set of observations.
+
+        Parameters
+        ----------
+        source_position : `~astropy.coordinates.SkyCoord`
+            Coordinates of the source position for which we want to calculate the psf
+        energy : `~astropy.units.Quantity`
+            1D Energy Array on which you want to define the new `~gammapy.irf.EnergyDependentTablePSF`
+        theta : `~astropy.coordinates.Angle`
+            1D Angle Array on which you want to define the new `~gammapy.irf.EnergyDependentTablePSF`
+
+        Returns
+        -------
+        energy_dependant_psftab : `~gammapy.irf.EnergyDependentTablePSF`
```

Comment:
Rename `energy_dependant_psftab` -> `psf`.
(that name just duplicates the info that is given by the type `EnergyDependentTablePSF` on the same line)